python/pyspark/ml/feature.py

```diff
@@ -1743,22 +1745,25 @@ def setStopWords(self, value):
         """
         Specify the stopwords to be filtered.
         """
-        self._paramMap[self.stopWords] = value
+        self._set(stopWords=value)
         return self
 
     @since("1.6.0")
     def getStopWords(self):
         """
         Get the stopwords.
         """
-        return self.getOrDefault(self.stopWords)
+        stopwords = self.getOrDefault(self.stopWords)
+        if isinstance(stopwords, JavaArray):
```

Comment:
This should be handled when the Param is set, not in the getter.  I wrote a little patch to test this.  Since we may want to backport the patch, I'll send it as a separate PR and CC you on it.  We can merge it quickly and then get yours merged afterwards.